helpers/network-recorder.js

```diff
@@ -21,14 +21,20 @@ const NetworkManager = require('./web-inspector').NetworkManager;
 const REQUEST_FINISHED = NetworkManager.EventTypes.RequestFinished;
 
 class NetworkRecorder {
-  constructor(recordArray) {
+  constructor(recordArray, requestListener) {
     this._records = recordArray;
 
+    this._listener = requestListener;
+
     this.networkManager = NetworkManager.createWithFakeTarget();
 
     // TODO(bckenny): loadingFailed calls are not recorded in REQUEST_FINISHED.
     this.networkManager.addEventListener(REQUEST_FINISHED, request => {
       this._records.push(request);
+
+      if (this._listener) {
```

Comment:
Just FYI, this is all changing to account for the different times chrome
may automatically fetch the manifest and then ignore the `requestAppBanner`
call (and/or request it twice), which was the cause of at least some of the
bugs in the initial comment.

On Tue, Apr 5, 2016, 18:27 Sam Saccone <notifications@github.com> wrote:

> In helpers/network-recorder.js
> <https://github.com/GoogleChrome/lighthouse/pull/137#discussion_r58640355>
> :
>
> >      this.networkManager = NetworkManager.createWithFakeTarget();
> >
> >      // TODO(bckenny): loadingFailed calls are not recorded in REQUEST_FINISHED.
> >      this.networkManager.addEventListener(REQUEST_FINISHED, request => {
> >        this._records.push(request);
> > +
> > +      if (this._listener) {
>
> +1 to an event emitter pattern vs an explicit callback in the constructor
> since it would be trivial to forget that a listener is set on an instance
> and experience a lot of frustration around why it is not working.
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/lighthouse/pull/137/files/f850cdb3e17c2fddbe571c8e56670ca97b7484e1#r58640355>
>